airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -255,7 +287,7 @@ def consume_logs(*, since_time: Optional[DateTime] = None, follow: bool = True)
                 time.sleep(1)
 
     def await_container_completion(self, pod: V1Pod, container_name: str) -> None:
-        while not self.container_is_running(pod=pod, container_name=container_name):
+        while self.container_is_running(pod=pod, container_name=container_name):
```

Comment:
that is the change unrelated to the issue.
I think the logic was reversed, blocking until the container is running (instead of blocking til terminated).

Reply:
It does look suspiciously wrong  indeed. Since this is used in your fix, I see no problem with having it as part of the PR.